be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -45,6 +45,48 @@ final case class DbActor(
     }
   }
 
+  @throws[DbActorNAckException]
+  private def writeCreateLao(channel: Channel, message: Message): Unit = {
+    createChannel(channel, ObjectType.LAO)
+    storage.write((storage.DATA_KEY + storage.CREATE_LAO_KEY + channel.toString, message.message_id.toString()))
+    write(Channel.ROOT_CHANNEL, message)
+  }
+
+  @throws[DbActorNAckException]
+  private def readCreateLao(channel: Channel): Option[Message] = {
```

Comment:
Now that I think about it, maybe the `read` function should be a bit commented (used for all messages except the CreateLao ones). It seems obvious since we add a new function `readCreateLao`, but some new students could still miss it. :)